packages/strapi-provider-upload-custom/src/index.ts

```diff
@@ -0,0 +1,29 @@
+import ProviderLocal from '@strapi/provider-upload-local';
+import { getAbsoluteServerUrl } from '@strapi/utils';
+
+const prefixFileUrlWithBackendUrl = (fileURL?: string): string | undefined => {
+  const baseUrl = getAbsoluteServerUrl(strapi.config);
+  return !!fileURL && fileURL.startsWith('/')
+    ? `${baseUrl}${fileURL}`
+    : fileURL;
+};
+
+export const init: typeof ProviderLocal.init = (options) => {
+  const fns = ProviderLocal.init(options);
+
+  return {
+    ...fns,
+    async uploadStream(file) {
+      // eslint-disable-next-line functional/no-expression-statements
+      await fns.uploadStream(file);
+      // eslint-disable-next-line functional/no-expression-statements, functional/immutable-data, @typescript-eslint/no-non-null-assertion
```

Comment:
**suggestion (non-blocking)**: If some rules doesn't make sense in this package you can override them ([here](https://github.com/pagopa/developer-portal/blob/main/packages/eslint-config/eslint-functional.js#L30-L34) an example)